src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/OpenSslX509ChainProcessor.cs

```diff
@@ -539,6 +539,28 @@ private static X509ChainStatusFlags MapVerifyErrorToChainStatus(Interop.Crypto.X
                         }
                     }
                 }
+
+                var candidatesByReference = new HashSet<X509Certificate2>(
+                    candidates,
+                    ReferenceEqualityComparer<X509Certificate2>.Instance);
+
+                // Dispose any certificates we cloned in, but didn't end up needing.
+                // Since extraStore was provided by users, don't dispose anything it contains.
+                Debug.Assert(storesToCheck.Length > 0, "storesToCheck.Length > 0");
+                Debug.Assert(storesToCheck[0] == extraStore, "storesToCheck[0] == extraStore");
+
+                for (int i = 1; i < storesToCheck.Length; i++)
```

Comment:
We skip the first one because it was provided by the user; they still (might) have references to the objects.  (The comments and asserts are about the "store 0 is special"ness)